### ISSUE TITLE
Serialization of DateTime loses Kind and TimeZone

### ISSUE THREAD

**Issue by MikaelUmaN** [Zaid-Ajaj/Fable.Remoting#273]
Hi.

This should be related to: https://github.com/Zaid-Ajaj/Fable.SimpleJson/pull/21 and https://github.com/Zaid-Ajaj/Fable.Remoting/issues/108

**Setup**
- dotnet new SAFE
- Create API accepting DateTime parameter
- From client, call server api with DateTime.Today (or DateTime.UtcNow.Date, ...)

**Expected**
- DateTime instance is the same on client and server.

**Actual**
- DateTime on client can be Kind=UTC (or Unspecified) but is always Local on the server.
- DateTime has Time component 00:00 (because we explicitly use the Date part), but when deserializing it becomes e.g. 2021-10-11 22:00 in my container.

**Why this is a problem for me**
Basically I am on dotnet 5 and eagerly awaiting the "DateOnly" struct.

As it is now, I expect to be able to send string(myDateParameter) to an external http API on the server-side but instead of todays date (e.g. 2021-10-12) I get yesterdays.

**Suggestions**
Probably it's not possible to keep the "Kind" type across serialisation, but then I think it should explicitly call "ToUniversalTime" prior to serialization and explicitly read it as Kind=UTC on the server side when deserializing.

The the application layer can handle the rest.

**Comment by Zaid-Ajaj** [comment]
Hi @MikaelUmaN,

In order not to lose the time zone information it is recommended to use `DateTimeOffset` which preserves the timezone offset of the server and propagates it to the client (that is represented by `Date`).

As for the Kind, I believe we had that implemented before but I need to double check
